### PR TITLE
Add inet datatype

### PR DESCRIPTION
**Pull request by pythonbrad**
### News
- Added INET datatype
- Added to_u32 and to_u128 for the data/bigdecimal_ext
### Linked Issues
- #814

### CODE REVIEW

**Comment by pythonbrad** [comment]
I think that this pull request is ready for review. ✍🏿

File: core/src/data/value/convert.rs [hunk 1]
```diff
@@ -513,26 +526,55 @@ impl TryFrom<&Value> for Interval {
...
     use {
         super::{parse_uuid, Value, ValueError},
         crate::{data::Interval as I, result::Result},
         chrono::{self, NaiveDate, NaiveDateTime, NaiveTime},
         rust_decimal::Decimal,
-        std::collections::HashMap,
+        std::{
+            collections::HashMap,
+            net::IpAddr,
+            net::{Ipv4Addr, Ipv6Addr},
```
Comment (panarch) [inline core/src/data/value/convert.rs:575]: nit. we can merge this `net::{IpAddr, Ipv4Addr, Ipv6Addr}`

**Review by panarch — CHANGES_REQUESTED** [review]
Thanks a lot 👍 
All the implementation codes look great.

Before merging this
could you add some more unit tests?

image
image

It is enough to see the file lists in `Changes Missing Coverage` and the lines which are marked as `NEW`.

**Review by panarch — APPROVED** [review]
Looks all great 👍 👍 👍 
So pleased to see the new data type :D 
Thanks a lot!

**Pull request merged**